redbeat/schedulers.py

```diff
@@ -39,6 +41,35 @@
 CELERY_4_OR_GREATER = CELERY_VERSION[0] >= 4
 
 
+class RetryingConnection(object):
+    """A proxy for the Redis connection that delegates all the calls to
+    underlying Redis connection while retrying on connection or time-out error.
+    """
+    RETRY_MAX_WAIT = 3
+
+    def __init__(self, retry_period, wrapped_connection):
+        self.retry_period = retry_period
+        self.wrapped_connection = wrapped_connection
+
+    def __getattr__(self, item):
+        method = getattr(self.wrapped_connection, item)
+
+        # we don't want to deal attributes or properties
+        if not callable(method):
+            return method
+
+        @retry(retry=(retry_if_exception_type(redis.exceptions.ConnectionError)
+                      | retry_if_exception_type(redis.exceptions.TimeoutError)),
+               reraise=True,
+               stop=(stop_after_delay(self.retry_period)),
+               wait=wait_fixed(min(self.retry_period // 2, self.RETRY_MAX_WAIT)))
```

Comment:
Is there a reason to use wait_fixed rather than wait_exponential or wait_random_exponential?  https://github.com/jd/tenacity#waiting-before-retrying